querydsl-examples/querydsl-example-jpa-guice/src/test/java/com/querydsl/example/jpa/repository/TweetRepositoryTest.java

```diff
@@ -75,4 +75,25 @@ public void find_list_by_complex_predicate() {
 
         assertEquals(100, repository.findAll(tweet.mentions.any().username.eq("dr_frank")).size());
     }
+
+    @Test
+    public void find_list_by_complex_predicate_hibernate() {
+        List<String> usernames = Lists.newArrayList("dr_frank", "mike", "maggie", "liza");
+        List<User> users = Lists.newArrayList();
+        for (String username : usernames) {
+            users.add(userRepository.save(new User(username)));
+        }
+        User poster = new User("duplo");
+        userRepository.save(poster);
+        for (int i = 0; i < 100; ++i) {
```

Comment:
Why ++i? Unconventional.

Reply:
Agreed. Took the tests as-is from an open issue on this bug.